### PR TITLE
[WIP] Add Alpine linux to the GitLab CI pipeline

### PR DESCRIPTION
**Pull request by D4N**
This required some refactoring of the scripts in `contrib/vms/` as synced folders with vagrant are broken on alpine linux. There is now a little duplication in the scripts, but I think it is minimal and still bearable.

Unfortunately, the tests are broken on alpine, as they use busybox' diff, which has fewer features then gnu diff. Our testsuite relies on the option `ignore-matching-lines=^` to exist, which doesn't in busybox.

### CODE REVIEW

**Review by piponazo — DISMISSED** [review]
Great! 

I was actually trying to do the same yesterday with Vagrant :grin: .

**Comment by piponazo** [comment]
There is a failure on the new image:

```
/bin/sh: eval: line 70: ci/install_dependencies.sh: not found
Uploading artifacts...
```
Weird ... I have approved taking into account that you will merge only once the CI checks are green.

**Comment by D4N** [comment]
Yeah, I think the alpine image has been seriously stripped down, so that even a shell is missing (then again it's only a few megabytes!). I'll try to fix that and won't merge this before that.

**Comment by clanmills** [comment]
Alpine offers several shell/environments.  I selected "busy-box" because I've used it a lot on embedded controllers.  It has a "OK" shell, however it's not bash.  It's an OK shell.  My understanding is that the shell/command-interpreter has most of the linux utilities built in.  Bash has builtin's such as cd, echo and so on.  Builtins are fast because you don't need to fork a process.  So it has a smaller run-time foot-print.  And it start new command very quickly.

I didn't have any difficulty getting Alpine installed and building Exiv2.  However you may need a special script to install the dependencies on Alpine.

If this is troublesome (or you're too busy to work on this), I'll get Jenkins to build on Alpine via ssh.  I expect that to be straightforward because the Alpine VM is already configured.

**Comment by D4N** [comment]
> Alpine offers several shell/environments.

Yeah, unfortunately one seems to need one for gitlab CI to run, but you can't use gitlab CI to select one, if it can't run. I guess there's a simple fix, we can't be the only people using the Alpine image from Dockerhub on Gitlab.

**Comment by D4N** [comment]
The tests are now passing, please don't merge this yet, I want to rebase the commits first.

**Comment by D4N** [comment]
@piponazo Could you take a look again, I had to do some tiny code changes.

File: unitTests/CMakeLists.txt
Comment (piponazo) [inline unitTests/CMakeLists.txt]: Good to know. I used GTest for years, and I never took the time to check whether gtest_main was needed or not.

**Comment by piponazo** [comment]
Everything is looking fine for me ;)

File: unitTests/CMakeLists.txt
Comment (D4N) [reply unitTests/CMakeLists.txt]: You can use `GTest::Main`, then you don't have to provide a `main()` that runs all tests yourself. But that requires that `gtest_main.a/.so` exists, which it doesn't on Alpine, so I'd prefer to keep it the way it is.

Comment (piponazo) [reply unitTests/CMakeLists.txt]: Sure. What I meant is that I always include it without thinking what it was. I actually always provide my own main function, specially in big projects where you need to do advance stuff ;)

**Review by piponazo — APPROVED** [review]

**Pull request merged**
